compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/PositionUtil.java

```diff
@@ -81,4 +81,19 @@ static boolean withinRange(LineRange specifiedRange, Location nodePosition) {
 
         return specifiedEndColumn <= nodeEndColumn;
     }
+
+    //  todo to be removed once https://github.com/ballerina-platform/ballerina-lang/discussions/28983 is sorted
+    static boolean withinRightInclusive(LinePosition cursorPos, Location symbolPosition) {
+        int startLine = symbolPosition.lineRange().startLine().line();
+        int startColumn = symbolPosition.lineRange().startLine().offset();
+        int endLine = symbolPosition.lineRange().endLine().line();
+
+        int cursorLine = cursorPos.line();
+        int cursorColumn = cursorPos.offset();
+        return (startLine < cursorLine && endLine > cursorLine)
+                || (startLine < cursorLine && endLine == cursorLine)
+                || (startLine == cursorLine && endLine > cursorLine)
```

Comment:
Is this check correct? What if the cursor is on the start line of the node, but before the start column? (Or this is correct based on the writing pattern?)

Reply:
Updated in 54acb97631c9572295a6e6dcc3a00b66e5add39f